sbndaq-artdaq/Generators/Common/CAENV1730Readout_generator.cc

```diff
@@ -1332,19 +1332,73 @@ bool sbndaq::CAENV1730Readout::checkHWStatus_(){
+      retcod = CAEN_DGTZ_ReadTemperature(fHandle, ch, &(ch_temps[ch]));
+      TLOG_ARB(TTEMP,TRACE_NAME) << tempStream.str()
+				 << ": " << ch_temps[ch] << "  C"
+				 << TLOG_ENDL;
+      
+      metricMan->sendMetric(tempStream.str(), int(ch_temps[ch]), "C", 11,
+			    artdaq::MetricMode::Average);
+
+      //Need 3 requirements to shut down for high temperature: 
+      // 1. Can successfully reead temperature: return code = 0 since S/N 164 can fail to read temperature during acquisition
+      // 2. Temperature > Requirement , 70C for V1730 and 85C for V1730S
+      // 3. Temperature < non_physical temperature, 200C since S/N 164 can produce non-physical temperature
+      if ( retcod == CAEN_DGTZ_Success )
+      {  
+	if ( ch_temps[ch] > fCAEN.maxTemp ) 
+	{
+	  retcod = CAEN_DGTZ_ReadTemperature(fHandle, ch, &(ch_temps[ch]));
+	  TLOG_ARB(TTEMP,TRACE_NAME) << tempStream.str()
+				     << ": " << ch_temps[ch] << "  C"
+				     << TLOG_ENDL;
+	  
+	  metricMan->sendMetric(tempStream.str(), int(ch_temps[ch]), "C", 11,
+				artdaq::MetricMode::Average);
+	  
+	  //Need 3 requirements to shut down for high temperature: 
+	  // 1. Can successfully reead temperature: return code = 0 since S/N 164 can fail to read temperature during acquisition
+	  // 2. Temperature > Requirement , 70C for V1730 and 85C for V1730S
+	  // 3. Temperature < non_physical temperature, 200C since S/N 164 can produce non-physical temperature
+	  if ( ch_temps[ch] < V1730_UNPHYSICAL_TEMPERATURE )
+	  {
+	    // V1730(S) shuts down at 70(85) celsius, give a warning ahead of that
+	    TLOG(TLVL_ERROR) << "SHUTTING DOWN CAENV1730 BoardID " << fBoardID << " : "
+			     << " Channel " << ch
+			     << " temperature " << ch_temps[ch]
+			     << " > " << fCAEN.maxTemp << " degrees Celsius."
+			     << " ReadTemperature Return Code = " << retcod
+			     << TLOG_ENDL;
+	  }
+	  else
+	  {
+	    // Ignore unphysical temperatures from S/N 164.  CAEN advises not to read temperatures 
+	    //   while the readout is running, but we cannot do that.  Only one sensors on one 
+	    //   V1730 has ever malfunctioned.
+	    // S/N 164 sometimes returns a non-physical temperature, ignore it and move on
+	    TLOG(TLVL_WARNING) << "CAENV1730 BoardID " << fBoardID << " : "
+			       << " Channel " << ch
+			       << " unphysical temperature " << ch_temps[ch] << " degrees Celsius."
+			       << " ReadTemperature Return Code = " << retcod
+			       << TLOG_ENDL;
+	  }
+	}
+      }
+      else
+      {
+	// Ignore unphysical temperatures from S/N 164.  CAEN advises not to read temperatures 
+	//   while the readout is running, but we cannot do that.  Only one sensors on one 
+	//   V1730 has ever malfunctioned.
+	TLOG(TLVL_WARNING) << "CAENV1730 BoardID " << fBoardID << " : "
+			       << " Channel " << ch
+			       << " temperature readout error " << ch_temps[ch] << " degrees Celsius."
+			       << " ReadTemperature Return Code = " << retcod
+			       << TLOG_ENDL;
+      }
```

Comment:
I'm quite confused by these nested `if` statements. I think they can be greatly simplified... there is no reason for example to read the temperature and send it to Grafana twice. Here is my suggestion:

```suggestion
      if ( retcod == CAEN_DGTZ_Success && ch_temps[ch] < V1730_UNPHYSICAL_TEMPERATURE )
      {  
	if ( ch_temps[ch] > fCAEN.maxTemp ) 
	{
	   // V1730(S) shuts down at 70(85) celsius, give a warning ahead of that
	   TLOG(TLVL_ERROR) << "SHUTTING DOWN CAENV1730 BoardID " << fBoardID << " : "
			     << " Channel " << ch
			     << " temperature " << ch_temps[ch]
			     << " > " << fCAEN.maxTemp << " degrees Celsius."
			     << " ReadTemperature Return Code = " << retcod
			     << TLOG_ENDL;
	}
      }
      else
      {
	    // Ignore unphysical temperatures/bad return codes from S/N 164.  CAEN advises not to read temperatures 
	    //   while the readout is running, but we cannot do that.  Only one sensors on one 
	    //   V1730 has ever malfunctioned.
	    // S/N 164 sometimes returns a non-physical temperature, ignore it and move on
	    TLOG(TLVL_WARNING) << "CAENV1730 BoardID " << fBoardID << " : "
			       << " Channel " << ch
			       << " unphysical temperature " << ch_temps[ch] << " degrees Celsius."
			       << " ReadTemperature Return Code = " << retcod
			       << TLOG_ENDL;
       }
```

I left the line sendng the temperatures to Grafana outside of the unphysical temperature check. I believe it can be a useful way to spot issues with the boards, but I won't complain too much if you prefer to have it inside the `if`.